### PR TITLE
Msis

### PR DESCRIPTION
**Pull request by joakimmoller**
Msis:
- pyarrow interprets DATE column as datetime.
- removed the "static" flux variant of extract profile

Start L2:
- Use StepFunctionsStartExecution task to start L2
- Map State and Pass to tweek to the right input

### CODE REVIEW

File: odincal/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,10 @@ RUN pip install -r requirements.txt
 
 # Install oops and odincal
 COPY ./oops /oops
+WORKDIR /oops/oops
+RUN gcc -shared -o libfft.so -fPIC ../Library/odinfft.c
```
Comment (skymandr) [inline odincal/Dockerfile:22]: What's the effect of this?

File: stacks/odincal_stack.py [hunk 2]
```diff
@@ -781,101 +780,53 @@ def set_up_create_zpt(
...
-        activate_level2_lambda = Function(
+        map_state = sfn.Map(
             self,
-            "OdinSMROdincalActivateLevel2Lambda",
-            code=InlineCode.from_asset("activate_l2"),
-            handler="handler.activate_l2_handler.activate_l2_handler",
-            timeout=lambda_timeout,
-            architecture=Architecture.X86_64,
-            runtime=Runtime.PYTHON_3_10,
+            "OdincalArrangeL2",
```
Comment (skymandr) [inline stacks/odincal_stack.py]:
```suggestion
            "OdinSMROdincalArrangeL2",
```
Alla andra heter så.

File: stacks/odincal_stack.py [hunk 3]
```diff
@@ -781,101 +780,53 @@ def set_up_create_zpt(
...
-            result_path="$.ActivateLevel2",
-        )
-        activate_level2_task.add_retry(
-            errors=["States.ALL"],
-            max_attempts=42,
-            backoff_rate=2,
-            interval=Duration.minutes(6),
-            max_delay=Duration.minutes(42),
-            jitter_strategy=sfn.JitterType.FULL,
+            "OdincalArrangeL2part",
```
Comment (skymandr) [inline stacks/odincal_stack.py]:
```suggestion
            "OdinSMROdincalArrangeL2part",
```

File: stacks/odincal_stack.py [hunk 4]
```diff
@@ -781,101 +780,53 @@ def set_up_create_zpt(
...
+        acticate_l2 = tasks.StepFunctionsStartExecution(
             self,
-            "OdinSMRActivateLevel2Status",
-        )
-
-        check_activate_level2_status_state.when(
-            sfn.Condition.number_equals(
-                "$.ActivateLevel2.Payload.StatusCode",
-                200,
+            "OdincalStartL2",
```
Comment (skymandr) [inline stacks/odincal_stack.py]:
```suggestion
            "OdinSMROdincalStartL2",
```

**Review by skymandr — CHANGES_REQUESTED** [review]
Looks good, but I'm wondering what the effect of adding the compile step in odin-cal is. Does this mean all data up until now needs to be re-run? We were planning to do so anyway, but I'd like to know.

Also, don't change naming standards for the Lambda's etc. without changing throughout. That just invites confusion.

File: odincal/Dockerfile (hunk 1 above)
Comment (joakimmoller) [reply odincal/Dockerfile:22]: Since there is no github action to deploy yet, I guess you already have the lib pre compiled in your working repo?

**Comment by joakimmoller** [comment]
> Looks good, but I'm wondering what the effect of adding the compile step in odin-cal is. Does this mean all data up until now needs to be re-run? We were planning to do so anyway, but I'd like to know.

Probably no effect, compared with precompiled in repo.

**Review by skymandr — APPROVED** [review]

**Pull request merged**